cachito/web/validation.py

```diff
@@ -46,3 +49,55 @@ def validate_dependency_replacements(replacements: List[Dict[str, Any]]) -> None
         raise ValidationError('"dependency_replacements" must be an array')
     for replacement in replacements:
         validate_replacement(replacement)
+
+
+class RequestBodyValidator(decorators.validation.RequestBodyValidator):
+    """
+    This class overrides the default connexion RequestBodyValidator.
+
+    It returns the complete string representation of the
+    error, rather than just returning the error message.
+
+    For more information:
+        - https://github.com/zalando/connexion/issues/558
+        - https://connexion.readthedocs.io/en/latest/request.html
+    """
+
+    def validate_schema(self, data, url):
+        """Throw cachito.ValidationError."""
+        if self.is_null_value_valid and jsonschema.is_null(data):
+            return None
+        try:
+            self.validator.validate(data)
+        except Exception as exception:
+            raise ValidationError(exception.message)
+
+        return None
+
+
+class ParameterValidator(decorators.validation.ParameterValidator):
+    """
+    This class overrides the default connexion ParameterValidator.
+
+    It returns the complete string representation of the
+    error, rather than just returning the error message.
+
+    For more information:
+        - https://github.com/zalando/connexion/issues/558
+        - https://connexion.readthedocs.io/en/latest/request.html
+    """
+
+    def __call__(self, function):
+        """Throw cachito.ValidationError."""
+        wrapper = super().__call__(function)
+
+        def handle_wrapper(request):
+            """Handle original wrapper."""
+            try:
+                return wrapper(request)
+            except Exception as e:
```

Comment:
I think it would be clearer here if we caught only the specific Connexion exceptions here (ExtraParameterProblem or BadRequestProblem) instead of checking for existence of a `details` attribute.

Any non-Connexion exception would be re-raised, as it is currently done here.

Reply:
If I make a request such as `api/v1/requests/2?hello=1`, I still get:
```
{
"detail": "Extra query parameter(s) hello not in spec",
"status": 400,
"title": null,
"type": "about:blank"
}
```
There's no unit test that covers extra URL query params, maybe we should add one.